build.sh

```diff
@@ -157,7 +157,19 @@ check_uuid_binary(){
 }
 
 run_build() {
-  /home/shippable/cexec/dist/main/main
+  local use_default_arch=true
+  if [ ! -z "$SHIPPABLE_NODE_ARCHITECTURE" ]; then
+    if [ -f /home/shippable/cexec/dist/"$SHIPPABLE_NODE_ARCHITECTURE"/linux/main/main ]; then
+      use_default_arch=false
+      /home/shippable/cexec/dist/"$SHIPPABLE_NODE_ARCHITECTURE"/linux/main/main
```

Comment:
- this should go in the `else` clause of `if use_default_arch==true`
- this block of code should just be concerned with setting the flag to true/false
- the next block will run the appropriate binary based on teh flag